subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoTaskExtension.java

```diff
@@ -234,6 +259,8 @@ public void setPort(int port) {
      *
      * @since 3.4
      */
+    @OutputDirectory
+    @Optional
```

Comment:
Is this an output or a local state?

Reply:
To me, it sounded more like an output: http://www.eclemma.org/jacoco/trunk/doc/agent.html
Maybe it is something like a temporary directory?

Reply:
Yeah, local temp directories are actually what `@LocalState` is for, no?

Reply:
`@LocalState` it is. Note that we declared this directory as an output via the runtime API in Gradle 4.5.